com.civicrm.hrjobroles/tests/phpunit/HrJobRolesTestBase.php

```diff
@@ -98,6 +108,27 @@ public function createSampleOptionGroupsAndValues()  {
       $params = ['option_group_id' => $groupID, 'name' => $value, 'value' => $value ];
       CRM_Core_BAO_OptionValue::create($params);
     }
+
+  }
+
+  /**
+   * Find and retrieve job role by any of its properties
+   *
+   * @param int|string $value
+   * @param string $identifier
+   *
+   * @return \CRM_Hrjobroles_BAO_HrJobRoles|NULL
+   */
+  public function findRoleByProperty($value, $identifier = 'id')  {
```

Comment:
Do we really need this method? I see that you made it possible use other properties instead of the ID, but  in every call you're always using the ID. You have the findByID method (inherited from CRM_Core_DAO) that does exactly the same thing.
To retrieve records matching other properties, you can use the commonRetrieve method, also inherited from CRM_Core_DAO